sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -154,8 +154,9 @@ fn type_check_encode_buffer_empty(
             TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
             TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
         ],
+        span.clone(),
     );
-    let return_type = type_engine.insert(engines, return_type, None);
+    let return_type = type_engine.insert(engines, return_type, span.source_id());
```

Comment:
Hmmm, I am not sure that this is the right solution, actually in this case `None` should be here. This is the very confusing thing about `source_id` parameter, it is not clear out of the parameter itself what should be given here.

The way I understood it, out of the usage and also out of our internal discussion at the time, `source_id` should be specified if the inserted type has a declaration that is in a particular source file, and then the `source_id` of that file should be specified, being a real file or autogenerated. Which is usually something like `decl.name.span().source_id()` or whichever `Span` is in the declaration file. For alias types, it's the file where the alias is declared etc.

Types like `Unknown`, `u64`, ... should have `source_id` set to `None`, as well as tuples combined only of those types, like above.

@tritao We will again need your help :smile:

Reply:
I reverted this change